source/gui/addonStoreGui/controls/storeDialog.py

```diff
@@ -268,6 +269,10 @@ def _requiresRestart(self) -> bool:
 			addonDataManager._downloadsPendingInstall
 			or state[AddonStateCategory.PENDING_INSTALL]
 		):
+			# Create a copy and merge the sets to log the add-ons pending install

```

Comment:
Wouldn't it be better to log these separately, so that if there is an unexpected value in one of these sets it is clear from where it comes from?

Reply:
good point